js/src/builtin/String.cpp

```diff
@@ -692,8 +690,8 @@ static bool str_unescape(JSContext* cx, unsigned argc, Value* vp) {
   }
 
   // TaintFox: add taint operation.
-  if(newtaint.hasTaint()) {
-    newtaint.extend(op);
+  if (newtaint.hasTaint()) {
+    newtaint.extend(TaintOperationFromContext(cx, "unescape", true, str));
```

Comment:
This could cause GC problems - if TaintOperationFromContext needs to allocate memory (e.g. by flattening str), then result might be freed as it is not rooted!

Reply:
Could probably mitigate this by moving it up before Step 6